pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/MailboxContentStreamObserver.java

```diff
@@ -116,11 +140,9 @@ public void onNext(Mailbox.MailboxContent mailboxContent) {
   @Override
   public void onError(Throwable e) {
     try {
-      _receivingBuffer.offer(Mailbox.MailboxContent.newBuilder()
-          .setPayload(ByteString.copyFrom(
-              TransferableBlockUtils.getErrorTransferableBlock(new RuntimeException(e)).getDataBlock().toBytes()))
-          .putMetadata(ChannelUtils.MAILBOX_METADATA_END_OF_STREAM_KEY, "true").build());
+      _errorContent = createErrorContent(e);
```

Comment:
this has to be atomic --> both onNext and onError can modify _errorContent

Reply:
Thanks for catching this. Fixed.